src/Common/Crypto/OpenSSLInitializer.cpp

```diff
@@ -31,22 +31,27 @@ void OpenSSLInitializer::initialize()
 #if USE_SSL
     if (ref_count++ == 0)
     {
+        // Disable OpenSSL atexit hook.
```

Comment:
If I read #80132 correctly, then we crashed when the atexit handler is installed, not when it runs at the end of the program. Also, https://docs.openssl.org/3.2/man3/OPENSSL_init_crypto/#description says that `OPENSSL_INIT_NO_ATEXIT By default OpenSSL will attempt to clean itself up when the process exits via an "atexit" handler. Using this option suppresses that behaviour. This means that the application will have to clean up OpenSSL explicitly using OPENSSL_cleanup().

My question is if we should call `OPENSSL_cleanup` in the dtor?

Reply:
`OPENSSL_cleanup` can be called only once according to their docs, so yes: we ideally call it during database shutdown and not here (dtor).